multiversion/src/main/scala/multiversion/commands/QueryBasedCommand.scala

```diff
@@ -0,0 +1,51 @@
+package multiversion.commands
+
+import java.nio.file.Path
+
+import moped.cli.Application
+import moped.cli.Command
+import moped.json.Result
+import moped.json.ValueResult
+import multiversion.diagnostics.MultidepsEnrichments._
+import os.Shellable
+
+/**
+ * A moped command that wraps around Bazel queries.
+ */
+abstract class QueryBasedCommand extends Command {
+  def app: Application
+  def bazel: Path
+  def bazelArgs: List[String]
+  def runCustomQuery(): Result[List[String]]
+
+  override def run(): Int = app.complete(runResult())
+
+  def runResult(): Result[Unit] =
+    for {
+      r <- runCustomQuery()
+    } yield r.foreach(Console.out.println(_))
```

Comment:
```suggestion
    } yield r.foreach(app.println(_))
```

I think moped swaps the streams in tests.